legend-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchJson.pure

```diff
@@ -192,51 +192,69 @@ function <<access.private>> meta::alloy::runtime::java::graphFetch::json::accept
-                         [                            
+                         [
                          $result->j_invoke('add', $acceptor->j_invoke('apply', $jsonNode, javaTypeVar('T')), javaList(javaTypeVar('T')))->j_try(j_catch($exception, $defectRecorder->j_invoke('accept', $exception->j_invoke('getMessage', [], javaString()), javaVoid())))
                          ]
                        ),
-                $result->j_return()     
+                $result->j_return()
               ]->codesToString()
             );
 }
 
-function <<access.private>> meta::alloy::runtime::java::graphFetch::json::createClassReadMethod(class:Class<Any>[1], proto:meta::java::metamodel::Class[1], path:String[1], context:GenerationContext[1]): meta::java::metamodel::Method[1]
+function <<access.private>> meta::alloy::runtime::java::graphFetch::json::createClassReadMethod(class:Class<Any>[1], proto:meta::java::metamodel::Class[1], path:String[1], context:GenerationContext[1], readableClasses:Class<Any>[*]): meta::java::metamodel::Method[1]
 {
    let conventions   = $context.conventions;
    let javaClass     = $conventions->className($class);
    let properties    = $context.typeInfos->valueProperties($class);
    let checked       = $conventions->className(Checked);
-   let checkedResult = javaParameterizedType($checked, $javaClass);   
+   let checkedResult = javaParameterizedType($checked, $javaClass);
 
    let jsonToken   = $conventions->codeType(javaClass('com.fasterxml.jackson.core.JsonToken'))->codeToString($conventions);
-   
+
+   let specializations = $class->meta::pure::functions::meta::findAllSpecializations()->filter(x|$x->in($readableClasses));
+   let nodeValueVariable = j_variable(javaString(), 'nodeValue');
+   let defectsVariable = j_variable(javaList($conventions->defectClass()), 'defects');
+   let specializationByName = $specializations->groupBy(x| $x.name->toOne());
+   let specializationsCode = if($specializations->isEmpty(), |'', |$specializationByName->keys()->toOneMany()
+                                ->map(className|pair(
+                                   $className->j_string()->j_invoke('equals', $nodeValueVariable),
+                                   if($specializationByName->get($className).values->size() == 1,
+                                        |$nodeValueVariable->j_assign(j_string($specializationByName->get($className).values->take(1)->toOne()->elementToPath('::'))),
+                                        {|
+                                           let defectMsg = j_string('multiple class matches [' + $specializationByName->get($className).values->map(c|$c->elementToPath('::')->escapedString())->sort()->joinStrings(', ') + '] for ' + $className->escapedString());
+                                           $defectsVariable->j_invoke('add', j_invoke($conventions->className(DataQualityBasicDefectClass), 'newInvalidInputCriticalDefect', [$defectMsg, j_string(escapedString($class->elementToPath()))]));
```

Comment:
Ahh yes!  I've removed the scape on this line.  I got confused from the other places defects are created, but I now notice that is because it concat text rather than using j_*** methods to generate the code.

Note that  I left thee escapes on line 240 as a mechanism to highlight the variables within the defect message to help users identify these.  Let me know if you think I should  remove the escape or a different way to highlight these.

Reply:
Those escapes should also be removed as they also end up as part of the concatenated result that is then wrapped in a j_string

Reply:
Ok, I have remove them on the last commit.